### PR TITLE
Updated swagger docs. Fixed minor bugs.

### PR DESCRIPTION
**Pull request by jacob-khoza-symb**
## Summary by CodeRabbit

- **New Features**
	- Added instructions for generating Swagger documentation in the README.

- **Improvements**
	- Updated API endpoint paths from "shlinks" to "share-links" for clarity and consistency.
	- Enhanced README formatting for better readability.
	- Restructured user profile information extraction from the authentication token for improved organization.

- **Refactor**
	- Streamlined code for QR code generation and session handling for better maintainability.

### CODE REVIEW

File: src/app/api/v1/share-links/[id]/qrcode/route.ts [hunk 1]
```diff
@@ -45,18 +38,20 @@ const shlinkRepo = container.get<ISHLinkRepository>(SHLinkRepositoryToken);
...
+) {
   try {
     const requestDto: SHLinkQRCodeRequestDto = await request.json();
     const { id } = params;
 
     let shlink = await getSingleSHLinkUseCase(
-      {repo: shlinkRepo},
-      {id: id, managementToken: requestDto.managementToken},
+      { repo: shlinkRepo },
+      { id: id, managementToken: requestDto.managementToken },
```
Comment (amalessid) [inline src/app/api/v1/share-links/[id]/qrcode/route.ts:51]: Hello @jacob-khoza-symb, 
A tiny thing here maybe if you want to simplify this
(based on the suggestion of the previous Comment)

```suggestion
      { id,managementToken },
```

**Review by godchiSymbionix — APPROVED** [review]

File: src/app/api/v1/share-links/[id]/route.ts
Comment (amalessid) [inline src/app/api/v1/share-links/[id]/route.ts:1]: Hello @jacob-khoza-symb, 

I suggest we simplify line 83 :  

  let requestDto: SHLinkRequestDto = await request.json();
=>   let { managementToken, passcode }: SHLinkRequestDto = await request.json();

which can help reduce the code in other lines.

File: src/app/utils/authentication.ts [hunk 2]
```diff
@@ -9,10 +9,20 @@ export interface UserProfile {
...
+  user: {
+    name: string;
+    id: string;
+    email: string;
+  };
+}
+
 export const getUserProfile = async (req: Request): Promise<UserProfile> => {
-  const token = await getToken({ req: req as NextRequest });
+  const token = (await getToken({
```
Comment (amalessid) [inline src/app/utils/authentication.ts:21]:
```suggestion
  const {user} = (await getToken({
```

File: src/app/utils/authentication.ts [hunk 3]
```diff
@@ -9,10 +9,20 @@ export interface UserProfile {
...
+    req: req as NextRequest,
+  })) as unknown as TokenPayload;
 
-  return { name: token.name, id: token.sub, email: token.email };
+  return { name: token.user.name, id: token.user.id, email: token.user.email };
```
Comment (amalessid) [inline src/app/utils/authentication.ts:25]:
```suggestion
  return { name: user.name, id: user.id, email: user.email };
```

File: src/app/api/v1/share-links/[id]/qrcode/route.ts [hunk 4]
```diff
@@ -45,18 +38,20 @@ const shlinkRepo = container.get<ISHLinkRepository>(SHLinkRepositoryToken);
...
  */
 
-export async function POST(request: Request, { params }: {params: { id: string } }) {
-    
+export async function POST(
+  request: Request,
+  { params }: { params: { id: string } },
+) {
   try {
     const requestDto: SHLinkQRCodeRequestDto = await request.json();
```
Comment (amalessid) [inline src/app/api/v1/share-links/[id]/qrcode/route.ts:46]:
```suggestion
    const { managementToken }: SHLinkQRCodeRequestDto = await request.json();
```

File: src/app/api/v1/share-links/[id]/qrcode/route.ts (hunk 1 above)
Comment (amalessid) [inline src/app/api/v1/share-links/[id]/qrcode/route.ts:51]:
```suggestion
      { id,managementToken },
```

**Review by amalessid — APPROVED** [review]

**Pull request merged**